### DISCUSSION TITLE
Deploying Blazor server-side; not displaying objects that have a per user restriction

### DISCUSSION THREAD

**Discussion by BlagoCuljak** [MarimerLLC/csla#1782] [Questions] [answered]
Hi, we are using Blazor 5.3, and Custom principal.

We have an object named Department, and some users have rights to add new Department, and this is done on clicking the button (if user has rights to add it, then the button is displayed:
`<AuthorizeView Policy="@(CslaPolicy.GetPolicy(AuthorizationActions.CreateObject, typeof(FactorOne.Library.DepartmentEdit)))">
        <Button RenderStyle="ButtonRenderStyle.Primary" Text="@Resources.Properties.Resources.Main_AddNew"
                  IconCssClass="Addition" Click="@AddNewClick" />
    </AuthorizeView>`

So far so good. The users that don't have the rights, they don't see that button. We can debug the app, and everything is fine, as far as we can see it.

However, when we release the code to IIS production server, the button is not displayed. We thought that the it must be something in the database that's different, and connect VS to production DB, and button is displayed. **And on production site is not, using same user.**

What do you think is wrong here? Not sure how to even debug server-side, since all API calls are SignalR related to UI:

**Comment by rockfordlhotka** [comment]
This is server-side Blazor?

Is the production server configured the same as in dev? Especially in terms of the web site authentication, etc.

**Reply by BlagoCuljak** [reply]
Well, in dev, we run it in debug IIS Express, straight from VS.

Let me try in the morning,and change those settings on IIS. 

Main issue with server side Blazor, how to debug production, when data API calls are not displayed on client...

**Reply by BlagoCuljak** [reply]
No difference, what-so-ever...

**Reply by rockfordlhotka** [reply]
Debugging SSB is pretty much the same as debugging MVC/Razor Pages - you need to attach a debugger to the server. Or implement robust logging. Just like the pre-SPA days of web development.

**Comment by rockfordlhotka** [comment, marked as answer]
Are you following the same authentication model as I walk through in my Blazor book? That technique relies on Blazor maintaining a consistent memory set on the server, which is how it should work.

**Reply by BlagoCuljak** [reply]
The entire Blazor solution (UI.Blazor and UI.Blazor.Server) was duplicated to our solution, with only their names changed.

**Reply by BlagoCuljak** [reply]
Thanks to uwe solutions, everything is looking good.

https://github.com/MarimerLLC/csla/discussions/1667
